src/modules/fancyzones/editor/FancyZonesEditor/MyButton.xaml.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Windows;
+using System.Windows.Automation;
+using System.Windows.Automation.Peers;
+using System.Windows.Automation.Provider;
+using System.Windows.Controls;
+
+namespace FancyZonesEditor
+{
+    public partial class MyButton : Button
+    {
+        public MyButton()
+            : base()
+        {
+            InitializeComponent();
+            Click += OnClick;
+        }
+
+        public string Value
+        {
+            get { return (string)GetValue(ValueProperty); }
+            set { SetValue(ValueProperty, value); }
+        }
+
+        public static readonly DependencyProperty ValueProperty =
+        DependencyProperty.Register(
+            "Value", typeof(string), typeof(MyButton));
+
+        private void OnClick(object sender, RoutedEventArgs e)
+        {
+            if (AutomationPeer.ListenerExists(AutomationEvents.PropertyChanged))
+            {
+                MyButtonAutomationPeer peer =
+                    UIElementAutomationPeer.FromElement(this) as MyButtonAutomationPeer;
+
+                if (peer != null)
+                {
+                    peer.RaisePropertyChangedEvent(
+                        ValuePatternIdentifiers.ValueProperty,
+                        null,
+                        Value);
+                }
+            }
+        }
+
+        protected override AutomationPeer OnCreateAutomationPeer()
+        {
+            return new MyButtonAutomationPeer(this);
+        }
+
+        public class MyButtonAutomationPeer : FrameworkElementAutomationPeer, IValueProvider
+        {
+            public MyButtonAutomationPeer(MyButton control)
+                : base(control)
+            {
+            }
+
+            protected override string GetClassNameCore()
+            {
+                return "MyButton";
```

Comment:
nameof(MyButton)?

Reply:
👍 MyButton will also be changed to something like ClickAutomationEventButton

Reply:
Perhaps we should use a more descriptive name rather than `MyButton`?

Reply:
> nameof(MyButton)?

Done

Also renamed MyButton to ClickAutomationEventButton